src/webui/api/apicontroller.cpp

```diff
@@ -105,3 +106,8 @@ void APIController::setResult(const QByteArray &result, const QString &mimeType,
     m_result.mimeType = mimeType;
     m_result.filename = filename;
 }
+
+void APIController::setStatus(const int statusCode)
+{
+    m_result.statusCode = statusCode;
+}
```

Comment:
This first commit is the same as the commit in #21015, since they both set custom status codes. I would like the API to be a better behaved REST API and use proper status codes going forward.